### DISCUSSION TITLE
Is there automatic serialization of JSON request bodies?

### DISCUSSION THREAD

**Discussion by jwodder** [sagebind/isahc#454] [Questions] [answered]
[The documentation for Isahc's `json` feature](https://docs.rs/isahc/latest/isahc/index.html#json) says that it enables (emphasis added):

> Additional **serialization** and deserialization of JSON bodies via [serde](https://serde.rs/).

While I can see how to deserialize JSON from a response using [`ReadResponseExt::json()`](https://docs.rs/isahc/latest/isahc/trait.ReadResponseExt.html#tymethod.json), I cannot find a parallel method for serializing JSON for a request (e.g., something like `reqwest`'s [`RequestBuilder::json()`](https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.json)), and [the JSON example](https://github.com/sagebind/isahc/blob/9d1edd475231ad5cfd5842d939db1382dc3a88f5/examples/json.rs) just performs serialization outside of Isahc, using `serde_json` directly.

Does Isahc provide or not provide a means to supply a `Serialize` value for the body of a request, with the conversion to JSON being perfomed by Isahc?

**Comment by sagebind** [comment, marked as answer]
Hi, thanks for your question! No, Isahc doesn't currently offer a method like that, but it is a good idea.
